controller/destination/server.go

```diff
@@ -286,6 +291,15 @@ func splitDNSName(dnsName string) ([]string, error) {
 		if l == "" {
 			return []string{}, errors.New("Empty label in DNS name: " + dnsName)
 		}
+		if !dnsCharactersRegexp.MatchString(l) {
+			return []string{}, errors.New("DNS name is too long or contains invalid characters: " + dnsName)
+		}
+		if !startsAndEndsWithAlphanumRegexp.MatchString(l) {
+			return []string{}, errors.New("DNS name cannot start or end with a dash: " + dnsName)
+		}
+		if !containsAlphaRegexp.MatchString(l) {
+			return []string{}, errors.New("DNS name cannot only contain numerals: " + dnsName)
```

Comment:
s/numerals/digits and hyphens/.

(The LHD rule uses "digit" instead of "numeral" to refer to digits.)